src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -121,7 +125,8 @@ object ReadModelQueries {
     val statesFilter       = mapToVarArgs(statesPartialFilter)(Filters.or)
     val eServicesIdsFilter = mapToVarArgs(eServicesIds.map(Filters.eq("data.id", _)))(Filters.or)
     val producersIdsFilter = mapToVarArgs(producersIds.map(Filters.eq("data.producerId", _)))(Filters.or)
-    val nameFilter         = name.map(Filters.regex("data.name", _, "i"))
+    val nameFilter         =
+      if (exactMatchOnName) name.map(Filters.eq("data.name", _)) else name.map(Filters.regex("data.name", _, "i"))
```

Comment:
in case of `exactMatchOnName == true` we lose the case insensitive match.
We want to avoid the creation of "Tax Service" along with "TaX SeRvIcE".

If you want to keep the boolean approach, the solution could be
```suggestion
      if (exactMatchOnName) name.map(n => Filters.regex("data.name", s"^$n$$", "i") else name.map(Filters.regex("data.name", _, "i"))
```
We still use the regex feature to apply case insensitive search, but the query now requires that there should not be any character before (with special character `^`) and after (with special character `$`) the given name.

Note that the double `$` is required as an escape to the string interpolation.